packages/material-ui/src/ClickAwayListener/ClickAwayListener.test.js

```diff
@@ -132,8 +132,7 @@ describe('<ClickAwayListener />', () => {
       expect(handleClickAway.callCount).to.equal(0);
 
       fireEvent.click(getByText('Stop inside a portal'));
-      // True-negative, we don't have enough information to do otherwise.
-      expect(handleClickAway.callCount).to.equal(1);
+      expect(handleClickAway.callCount).to.equal(0);
```

Comment:
See discussion in https://github.com/mui-org/material-ui/pull/20406#discussion_r467262774. Seems to me expected behavior matches desired behavior with `react@next`

Reply:
Awesome, React helps us here :) Looks all good: https://github.com/mui-org/material-ui/pull/20406#discussion_r467475291